scripts/installer.sh

```diff
@@ -0,0 +1,345 @@
+			# TODO: is that true? When was tailscale packaged?
+			;;
+		void)
+			# Rolling release, no version checking needed.
+			;;
+		gentoo)
+			# Rolling release, no version checking needed.
+			;;
+		freebsd)
+			if [ "$VERSION" != "12" ] && \
+			   [ "$VERSION" != "13" ]
+			then
+				OS_UNSUPPORTED=1
+			fi
+			;;
+		openbsd)
+			OS_UNSUPPORTED=1
+			;;
+		macos)
+			# We delegate macOS installation to the app store, it will
+			# perform version checks for us.
+			;;
+		other-linux)
+			OS_UNSUPPORTED=1
+			;;
+		*)
+			OS_UNSUPPORTED=1
+			;;
+	esac
+	if [ "$OS_UNSUPPORTED" = "1" ]; then
+		case "$OS" in
+			other-linux)
+				echo "Couldn't determine what kind of Linux is running."
+				echo "You could try the static binaries at:"
+				echo "https://pkgs.tailscale.com/stable/#static"
+				;;
+			"")
+				echo "Couldn't determine what operating system you're running."
+				;;
+			*)
+				echo "$OS $VERSION isn't supported by this script yet."
+		esac
+		echo
+		echo "If you'd like us to support your system better, please email support@tailscale.com"
+		echo "and tell us what OS you're running."
+		echo
+		echo "Please include the following information we gathered from your system:"
+		echo
+		echo "OS=$OS"
+		echo "VERSION=$VERSION"
+		echo "PACKAGETYPE=$PACKAGETYPE"
+		if [ -f /etc/os-release ]; then
+			cat /etc/os-release
+		else
+			echo "No /etc/os-release"
+		fi
+		if which uname >/dev/null; then
+			echo "uname: $(uname -a)"
+		else
+			echo "No uname"
+		fi
+		exit 1
+	fi
+
+	# Step 3: run the installation.
+	echo "Installing Tailscale for $OS $VERSION, using method $PACKAGETYPE"
+	case "$PACKAGETYPE" in
+		apt)
+			# TODO: use newfangled per-repo signature scheme
+			set -x
+			curl -fsSL "https://pkgs.tailscale.com/stable/$OS/$VERSION.gpg" | sudo apt-key add -
+			curl -fsSL "https://pkgs.tailscale.com/stable/$OS/$VERSION.list" | sudo tee /etc/apt/sources.list.d/tailscale.list
+			sudo apt-get update
+			sudo apt-get install tailscale
+			set +x
+		;;
+		yum)
+			set -x
+			sudo yum install yum-utils
+			sudo yum-config-manager --add-repo "https://pkgs.tailscale.com/stable/$OS/$VERSION/tailscale.repo"
+			sudo yum install tailscale
+			sudo systemctl enable --now tailscaled
+			set +x
+		;;
+		dnf)
+			set -x
+			sudo dnf config-manager --add-repo "https://pkgs.tailscale.com/stable/$OS/$VERSION/tailscale.repo"
+			sudo dnf install tailscale
+			sudo systemctl enable --now tailscaled
+			set +x
+		;;
+		zypper)
+			set -x
+			sudo zypper ar -g -r "https://pkgs.tailscale.com/stable/$OS/$VERSION/tailscale.repo"
+			sudo zypper ref
+			sudo zypper in tailscale
+			sudo systemctl enable --now tailscaled
+			set +x
+			;;
+		pacman)
+			sudo pacman -S tailscale
+			sudo systemctl enable --now tailscaled
+			;;
+		apk)
+			sudo apk add tailscale
+			sudo rc-update add tailscale
+			;;
+		xbps)
+			sudo xbps-install tailscale
+			;;
+		emerge)
+			sudo emerge net-vpn/tailscale
+			;;
+		appstore)
```

Comment:
This is cute, but is it what we want? CLI tailscaled does work on darwin. It might be better to print out a message instead, a la "Install our app here: https://... or install the CLI version here: ...".

Reply:
We currently don't have a distribution of our CLI version (or the sysext version), so I think this is something we'd add to the script once we're distributing them?

Reply:
@danderson's comment is a good point for the near future.

For the further future, I think we should treat one Mac build as canonical and always drive people to that. Maybe at some point in the future, the CLI or sysext builds will be canonical once we have auto-updating, more thorough testing, etc. But for now, the macOS App Store build is our canonical build, and it may be nicest to just use that.

Reply:
I mostly agree, but note that this script is (hopefully obviously) only for technical people who have decided to start running stuff from the command line, rather than doing the obvious thing and getting it from the app store. That's a strong, albeit not perfect, indicator that they don't want the app store version for some reason. (Two good reasons: it's possible to make CLI tailscaled start up before you log in; it's possible to upgrade/downgrade tailscaled in your MDM software and have complete control over which version you use, like you can on Linux/Windows.)

If we had a supported production-quality homebrew-style tailscaled package, I'd probably expect this script to install that. Since we don't, the most useful thing to me as a user would be to tell me that I've gone wrong and suggest the different alternatives.